src/main/solidity/greeter/Greeter.sol

```diff
@@ -1,40 +1,38 @@
-pragma solidity ^0.4.25;
-
-// Modified Greeter contract. Based on example at https://www.ethereum.org/greeter.
+pragma solidity >=0.4.22 <0.6.0;
 
 contract Mortal {
-    /* Define variable owner of the type address*/
+    /* Define variable owner of the type address */
     address owner;
 
-    /* this function is executed at initialization and sets the owner of the contract */
-    constructor () public { owner = msg.sender; }
+    /* This constructor is executed at initialization and sets the owner of the contract */
+    constructor() public { owner = msg.sender; }
 
     /* Function to recover the funds on the contract */
-    function kill() public { if (msg.sender == owner) selfdestruct(owner); }
+    function kill() public { if (msg.sender == owner) selfdestruct(msg.sender); }
 }
 
 contract Greeter is Mortal {
-    /* define variable greeting of the type string */
+    /* Define variable greeting of the type string */
     string greeting;
 
-    /* this runs when the contract is executed */
-    constructor (string _greeting) public {
+    /* This runs when the contract is executed */
+    constructor(string memory _greeting) public {
```

Comment:
what was the reason for adding the `memory` declaration to strings? As function parameters is this not already implicit? I'd like to keep the code as simple as possible for users unfamiliar to Solidity, so I'd rather keep the memory declaration out if we can.